tests/mean_04.cc

```diff
@@ -0,0 +1,125 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2019 by the SampleFlow authors.
+//
+// This file is part of the SampleFlow library.
+//
+// The SampleFlow library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of SampleFlow.
+//
+// ---------------------------------------------------------------------
+
+
+// Check the MeanValue consumer
+
+
+#include <iostream>
+#include <valarray>
+#include <random>
+
+#include <sampleflow/producers/metropolis_hastings.h>
+#include <sampleflow/consumers/mean_value.h>
+
+
+class MyTriangle
+
+{
+  public:
+
+    std::valarray<double> side_lengths;
+
+    MyTriangle()
+    {
+      side_lengths = {std::numeric_limits<double>::signaling_NaN(),
```

Comment:
I think you can do this in a more elegant way as part of the "member initializer list":
```
MyTriangle ()
  :
  side_lengths ({std::numeric_limits<double>::signaling_NaN(),
                          std::numeric_limits<double>::signaling_NaN(),
                          std::numeric_limits<double>::signaling_NaN())
 {}
```
I'll be happy to discuss why on Wed.

Reply:
Implemented, but yes I'd love to hear more about this on wednesday.

Reply:
Actually, you read about it in the book this week: Initialize where allocation happens. For objects in C++, allocation happens at object creation time, i.e., the constructor and the earliest you can initialize objects is in the member initializer list. But happy to talk more :-)